### ISSUE TITLE
Enhancement: callbacks for server start and stop

### ISSUE THREAD

**Issue by montsamu** [herrevilkitten/telnet-rxjs#9]
I'd prefer callbacks instead of promises, but you're the owner, so you pick, unless you want to wait for a pull request from me which will be callbacks. :) But I find myself wanting a callback from server.start() and server.stop() enough that I am tired of adding a one-time temporary listener for the Started and Ended events, and even as it is I don't like  at all what I end up with:

in MyBigServer:

```
  self.telnetd = Telnet.server(...)

  self.start = (cb) => {
    if (undefined !== cb) {
      self.telnetd.filter((event) => event instanceof Telnet.Event.Started).subscribe((event) => {
        if ( cb !== undefined ) {
          cb();
          cb = undefined; // TODO how to actually unsubscribe?!
        }
      });
    }
    return self.telnetd.start();
  }

  self.stop = (cb) => {
    self.telnetd.filter((event) => event instanceof Telnet.Event.Ended).subscribe((event) => {
      self.telnetd.server.unref();
      self.telnetd.complete();
      self.telnetd.unsubscribe();
      self.telnetd = undefined;
      if ( cb !== undefined ) {
        cb();
        cb = undefined; // TODO how to actually unsubscribe?!
      }
    });
    return self.telnetd.stop();
  }

```

So instead of doing most of that, I could just:

```

  self.telnetd.start(() => {
    // do a thing I want to do this time when telnetd starts
  });

  self.telnetd.stop(() => {
    self.telnetd.server.unref();
    self.telnetd.complete();
    self.telnetd.unsubscribe();
  });

```

Obviously I have my own room for improvement first, as I am not up to speed enough with rxjs to yet figure out "subscribe only for the first event of this type, then stop subscribing", but that's what I'm going for.

**Comment by herrevilkitten** [comment]
How would you feel about `EventEmitter`?

```
self.telnetd.on('start', () => {
   ...
});
```

**Comment by montsamu** [comment]
I might even prefer promises to EventEmitter in this case, but that would be a step in the right direction for me that I could live with. I should be able to do:

self.telnetd.once('start' ...)

and:

self.telnetd.once('stop' ...)

And... there's probably an EventEmitter wrapper for RxJS project already out there, isn't there.

**Comment by herrevilkitten** [comment]
After some thought, `EventEmitter` wouldn't work too well, since we're already inheriting from `ReplaySubject`.  So I'm thinking of changing `Server#start` and `Server#stop` to return a `Promise<net.Server | tls.Server>` and possibly also support an optional callback.

**Comment by herrevilkitten** [comment]
Fixed in v2.1.0

`start` and `stop` now return promises that resolve when the appropriate underlying method (`listen` / `close`) have completed.
